kafka_utils/util/metadata.py

```diff
@@ -42,16 +43,27 @@ def get_topic_partition_metadata(hosts):
     return topic_partitions
 
 
-def get_topic_partition_with_error(cluster_config, error):
+def get_topic_partition_with_error(cluster_config, error, fetch_unavailable_brokers=False):
     """Fetches the metadata from the cluster and returns the set of
     (topic, partition) tuples containing all the topic-partitions
-    currently affected by the specified error"""
+    currently affected by the specified error. It also fetches unavailable-broker list
+    if required."""
 
     metadata = get_topic_partition_metadata(cluster_config.broker_list)
     affected_partitions = set()
+    if fetch_unavailable_brokers:
+        unavailable_brokers = set([])
     for partitions in metadata.values():
         for partition_metadata in partitions.values():
             if int(partition_metadata.error) == error:
+                if fetch_unavailable_brokers:
```

Comment:
Can we have a separate function to get the unavailable brokers for a given partition_metadata and zk connection.

Also we can open the ZK connection outside of the loop so we don't keep creating new connections.

Reply:
Could be something like

```
def get_unavailable_brokers(partition_metadata, zk):
```